From c072c47ae19e1c106be34d088945cf99653b3bfc Mon Sep 17 00:00:00 2001
From: benhe2011 <benhe.che@gmail.com>
Date: Tue, 6 Aug 2019 17:30:42 -0700
Subject: [PATCH 3/3] add make script

---
 upsample_update | 6 ++++++
 1 file changed, 6 insertions(+)
 create mode 100755 upsample_update

diff --git a/upsample_update b/upsample_update
new file mode 100755
index 000000000..07265ca09
--- /dev/null
+++ b/upsample_update
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+make -j2 USE_OPENCV=1 USE_OPENMP=0 DEBUG=1 USE_PROFILER=1
+cd python
+pip install -e .
+cd ..
\ No newline at end of file
-- 
2.20.1 (Apple Git-117)

